packages/telemetry/src/flight-recorder.js

```diff
@@ -30,7 +30,53 @@ const I_CIRC_START = 2;
 const I_CIRC_END = 3;
 const HEADER_LENGTH = 4;
 
-export const makeMemoryMappedCircularBuffer = ({
+const initializeCircularBuffer = async (bufferFile, circularBufferSize) => {
+  if (!circularBufferSize) {
+    return undefined;
+  }
+  // If the file doesn't exist, or is not large enough, create it.
+  const stbuf = await fsPromises.stat(bufferFile).catch(e => {
+    if (e.code === 'ENOENT') {
+      return undefined;
+    }
+    throw e;
+  });
+  const arenaSize = BigInt(
+    circularBufferSize - HEADER_LENGTH * BigUint64Array.BYTES_PER_ELEMENT,
+  );
+
+  const writeHeader = async () => {
+    if (
+      stbuf &&
+      stbuf.size >= HEADER_LENGTH * BigUint64Array.BYTES_PER_ELEMENT
+    ) {
+      // Header already exists.
+      return;
+    }
+
+    // Write the header.
+    const header = new Array(HEADER_LENGTH).fill(0n);
+    header[I_MAGIC] = SLOG_MAGIC;
+    header[I_ARENA_SIZE] = arenaSize;
+    await fsPromises.mkdir(path.dirname(bufferFile), { recursive: true });
+    await fsPromises.writeFile(bufferFile, BigUint64Array.from(header));
+  };
+  await writeHeader();
```

Comment:
I can see the Jessie style peering through. I was wondering why the `if` was inside `writeHeader`. I need to get used to this!